lib/stormpath-sdk/resource/application.rb

```diff
@@ -118,20 +116,18 @@ def authenticate_oauth(request)
   private
 
   def jwt_token_payload(options)
-    payload = {
-      'iat' => Time.now.to_i,
-      'jti' => UUID.method(:random_create).call.to_s,
-      'iss' => client.data_store.api_key.id,
-      'sub' => href,
-      'cb_uri' => options[:callback_uri],
-      'path' => options[:path] || '',
-      'state' => options[:state] || '',
-    }
-
-    payload["sof"] = options[:show_organization_field] if options[:show_organization_field]
-    payload["onk"] = options[:organization_name_key] if options[:organization_name_key]
-    payload["usd"] = options[:use_subdomain] if options[:use_subdomain]
-    payload
+    {}.tap do |payload|
+      payload['iat'] = Time.now.to_i
+      payload['jti'] = UUID.method(:random_create).call.to_s
+      payload['iss'] = client.data_store.api_key.id
+      payload['sub'] = href
+      payload['cb_uri'] = options[:callback_uri]
+      payload['path'] = options[:path] || ''
+      payload['state'] = options[:state] || ''
+      payload['sof'] = options[:show_organization_field] if options[:show_organization_field]
+      payload['onk'] = options[:organization_name_key] if options[:organization_name_key]
+      payload['usd'] = options[:use_subdomain] if options[:use_subdomain]
+    end
```

Comment:
Btw one neat trick you can use so you don't have to use these `if's` is to use the `Hash#compact` method:

```ruby
  def jwt_token_payload(options)
    {}.tap do |payload|
      payload['iat'] = Time.now.to_i
      payload['jti'] = UUID.method(:random_create).call.to_s
      payload['iss'] = client.data_store.api_key.id
      payload['sub'] = href
      payload['cb_uri'] = options[:callback_uri]
      payload['path'] = options[:path] || ''
      payload['state'] = options[:state] || ''
      payload['sof'] = options[:show_organization_field]
      payload['onk'] = options[:organization_name_key]
      payload['usd'] = options[:use_subdomain]
    end.compact
end
```

The `Hash#compact` method will remove all key-value pairs that have a value of nil.